src/searched.js

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable  max-len, no-restricted-syntax, guard-for-in, quotes, no-dupe-keys, camelcase, no-trailing-spaces, no-plusplus,  import/no-cycle */
+import { Key } from './index';
+import { Logic } from './logic';
+
+class Search {
+  constructor(_name) {
+    this.name = _name;
+  }
+
+
+  getSearchedLocation() {
+    const cityName = this.name;
+
+    async function getLocation() {
+      const response = await fetch(
+
+
+        `https://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=${Key.theKey()}`,
+        {
+          mode: 'cors',
+        },
+
+      );
+
+      const weatherData = await response.json();
+
+
+      Logic.logicInfo(weatherData);
+    }
+
+    getLocation();
+  }
```

Comment:
- **_It should have error handling for `promises` or `async/await` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/error_handling_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))_**